app/forms/claims/claim_window_form.rb

```diff
@@ -0,0 +1,68 @@
+class Claims::ClaimWindowForm < ApplicationForm
+  attr_accessor :id
+
+  date_attribute :starts_on
+  date_attribute :ends_on
+
+  attribute :academic_year_id
+
+  validates :starts_on, presence: true
+  validates :ends_on, presence: true, comparison: { greater_than_or_equal_to: :starts_on }
+  validates :academic_year_id, presence: true
+
+  validate :is_not_longer_than_an_academic_year
+  validate :does_not_start_within_another_claim_window
+  validate :does_not_end_within_another_claim_window
+
+  def initialize(attributes = {})
+    super
+
+    if id
+      self.starts_on ||= claim_window.starts_on
+      self.ends_on ||= claim_window.ends_on
+      self.academic_year_id ||= claim_window.academic_year_id
+    end
+  end
+
+  def persist
+    claim_window.update!(starts_on:, ends_on:, academic_year_id:)
+  end
+
+  def new_back_path
+    new_claims_support_claim_window_path(claims_claim_window_form: slice(:starts_on, :ends_on, :academic_year_id))
+  end
+
+  def edit_back_path
+    edit_claims_support_claim_window_path(id, claims_claim_window_form: slice(:starts_on, :ends_on, :academic_year_id))
+  end
+
+  def academic_year_name
+    AcademicYear.find_by(id: academic_year_id)&.name
+  end
+
+  private
+
+  def claim_window
+    @claim_window ||= id ? Claims::ClaimWindow.find(id) : Claims::ClaimWindow.new
+  end
+
+  def is_not_longer_than_an_academic_year
+    return if starts_on.blank? || ends_on.blank?
+
+    if (ends_on - starts_on).to_i > 365
+      errors.add(:base, :longer_than_academic_year)
+    end
+  end
```

Comment:
You've removed this validation from the claim window model. I imagine it needs removing here too.

Reply:
> Allow over year-long windows at the Model level. Prevent over year-long windows at the FormObject level.

No, this is intentional. In the case that this behaviour is desired, it can be overridden by a developer.

Reply:
Whilst I take issue that a Developer would need to change this data, and that I believe if this is valid then a Support User should be trusted enough to extend a ClaimWindow without assistance from Developers.

If this is intentional I will concede.

However, this validation checks that a ClaimWindow is less than a whole year (even though a leap year is 366 days). A better check would be to check if the start and end dates are within the boundaries of an actual AcademicYear. 
Otherwise a SupportUser can make a ClaimWindow for 29 Aug 2024 - 28 Aug 2025, which doesn't seem like the desired outcome.

Reply:
I'll make this adjustment.

Reply:
I made a mistake here. This isn't quite right. A claim window does not need to be within the bounds of an academic year. A claim window for a given year will be opened after the academic year has ended.

Reply:
When then at the very least, there should be validation to ensure that the placement window begins within the bounds of an academic year

Reply:
That's also not quite right. There's no necessity for the window to begin, nor end, within the bounds of its associated academic year.

Reply:
So given your last comment "There's no necessity for the window to begin, nor end, within the bounds of its associated academic year." I would return to my original comment, that this validation should be removed, as this validation is not shared with the model this form is intended to represent. Support Users should "given what you've said" be able to create an ClaimWindow which branches over academic years.

This does lead me to question the necessary association of a ClaimWindow with an AcademicYear though.

Reply:
> Support Users should "given what you've said" be able to create an ClaimWindow which branches over academic years.

Arguably, yes. I can remove the limitation if you don't think it is reasonable.

It was introduced to help add sensible limits, similar to cases of not allowing people to add date of births before 1900.

> This does lead me to question the necessary association of a ClaimWindow with an AcademicYear though.

All claims made within a claim window should all be associated to the same academic year. This allows us to query claims against both claim windows _and_ academic years.

Reply:
![Screenshot 2024-08-21 at 16 03 03](https://github.com/user-attachments/assets/4dca510e-25f7-4023-8fb0-0a1ab9054632)

But given your current implementation I can still do ^

Reply:
In summary, I think the validations should reflect exactly what's on the model. The form shouldn't be adding additional validation, especially when it's been removed from the model itself given the existing use case that a claim window, can in fact, be longer than an year in length.

Reply:
In that case, let's remove the _sensibility_ validations, as it is likely hard to determine what is _actually_ sensible.

We'll rely on the Support team to decide how they want to schedule windows. Technically, we'll validate that the records are logically sound - such as the presence of fields, no overlap, and date ranges that start and end respectively.